### PR TITLE
Modify index.html and script.js to use fetch to request server content

### PR DESCRIPTION
**Pull request by amuamushu**
Screenshot of button: https://screenshot.googleplex.com/EFq0aYaA2J0.png
Screenshot of requested content appended to the page: https://screenshot.googleplex.com/yXTpP8tXGU1.png

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -119,3 +119,37 @@ function hideOnMouseout(hoveredItem) {
...
   }
 }
+
+/**
+ * Fetches the message from the server /data and adds it to the DOM.
+ */
+function getMessage() {
+  const responsePromise = fetch('/data');
+  
+  // Passes responsePromise into handleResponse() after
```
Comment (f-camp) [inline portfolio/src/main/webapp/script.js:129]: No need for this comment as the code explains itself. Use comments where the logic might be extremely complex. A lot of the times good method naming and separation of concerns helps in describing the code clearly.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -119,3 +119,37 @@ function hideOnMouseout(hoveredItem) {
...
+/**
+ * Handles response by converting it to text and passing the result
+       to addMessageToDom().
+ * @param {object} response A promise that was fetched from a URL.
+ */
+function handleResponse(response) {
+  const textPromise = response.text();
+
+  // Passes textPromise into addMessageToDom() after the 
+  // textPromise is converted to text.
```
Comment (f-camp) [inline portfolio/src/main/webapp/script.js:143]: ditto.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: Done!

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: Done!

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -83,10 +83,10 @@ function createPTag(text) {
  *     project image and text.
  */
 function revealOnMouseover(hoveredItem) {
-  let toReveal = hoveredItem.getElementsByClassName('hidden')[0];
-  let toHide = hoveredItem.getElementsByClassName('visible')[0];
+  const toReveal = hoveredItem.getElementsByClassName('hidden')[0];
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:86]: I would make hidden, background, visible all global consts.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -119,3 +119,33 @@ function hideOnMouseout(hoveredItem) {
...
+ * Fetches the message from the server /data and adds it to the DOM.
+ */
+function getMessage() {
+  const responsePromise = fetch('/data');
+  
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script.js:133]: I would rephrase this comment because now the user needs to look up what addMessageToDom does to understand what this function does. I would re-write it as: Obtains the reponse's text and adds it to the DOM.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js]: Done!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (amuamushu) [reply portfolio/src/main/webapp/script.js:139]: I believe this method only converts the response to text. The next method (addMessageToDom()) would add it to the DOM instead. Is it okay if I re-write it to say "Obtains the response's text" instead?
